entrypoint.sh

```diff
@@ -1,39 +1,34 @@
 #!/bin/bash
+DIR="$1"
+FILENAME="$2"
+URL="$3"
+APP_NAME="$4"
+BRANCH="$5"
 
-git config --global --add safe.directory "/github/workspace/$1"
-
-cd $1
-export DATA=$(curl $3 | jq --arg APP_NAME "$4" '{"data":map(select(.appName == $APP_NAME))}')
-
-if [[ $2 =~ .*\.json ]]; then
-    echo -e "$DATA" > $2
-# elif [[ $2 =~ .*\.swift ]]; then
-#     echo -e "// This Source Code Form is subject to the terms of the Mozilla Public\n// License, v. 2.0. If a copy of the MPL was not distributed with this\n// file, You can obtain one at http://mozilla.org/MPL/2.0/\n\nimport Foundation\n\npublic struct FirstRunExperiments {\n    public static let value = \"\"\"$DATA\"\"\"\n}" > $2
-# elif [[ $2 =~ .*\.kt ]]; then
-#     if [[ -z $6 ]]; then
-#         echo "Package is not specified. Package is a required value for the kotlin language."
-#         exit 1
-#     fi
-#     echo -e "// This Source Code Form is subject to the terms of the Mozilla Public\n// License, v. 2.0. If a copy of the MPL was not distributed with this\n// file, You can obtain one at http://mozilla.org/MPL/2.0/\n\npackage $6\n\n/*\n * This is a list of the first-run experiments. It will be passed to Nimbus during startup.\n */\nconst val firstRunExperiments = \"\"\"$DATA\"\"\"" > $2
-else
+if [[ ! $FILENAME =~ .*\.json ]]; then
     echo "Language of output file is not supported. Please set the output to a JSON file."
     exit 1
 fi
 
+git config --global --add safe.directory "/github/workspace/$DIR"
+pushd "$DIR" || exit 1
+
+curl "$URL" | jq --arg APP_NAME "$APP_NAME" '{"data":map(select(.appName == $APP_NAME))}' > "$FILENAME"
```

Comment:
This is the fix here. The rest is shellcheck tidying.

We could have used `echo -E`, or just `echo`. In the end, I removed the `echo` altogether.